aspnetcore/release-notes/aspnetcore-6.0.md

```diff
@@ -0,0 +1,682 @@
+
+#### Zero byte reads with PipeReader
+
+With bufferless reads supported on `SslStream`, an option was added to perform zero byte reads to `StreamPipeReader`, the internal type that adapts a `Stream` into a `PipeReader`. In Kestrel, a`StreamPipeReader` is used to adapt the underlying `SslStream` into a `PipeReader`. Therefore it was necessary to expose these zero byte read semantics on the `PipeReader`.
+
+A `PipeReader` can now be created that supports zero bytes reads over any underlying `Stream` that supports zero byte read semantics (e.g,. `SslStream`, `NetworkStream`, etc) using the following API:
+
+```dotnetcli
+var reader = PipeReader.Create(stream, new StreamPipeReaderOptions(useZeroByteReads: true));
+```
+
+#### Remove slabs from the `SlabMemoryPool`
+
+To reduce fragmentation of the heap, Kestrel employed a technique where it allocated slabs of memory of 128 KB as part of it’s memory pool. The slabs were then further divided into 4 KB blocks that were used by Kestrel internally. The slabs had to be larger than 85 KB to force allocation on the large object heap to try and prevent the GC from relocating this array. However, with the introduction of the new GC generation, Pinned Object Heap (POH), it no longer makes sense to allocate blocks on slab. In preview3, we now directly allocate blocks on the POH, reducing the complexity involved in managing our own memory pool. This change should make easier to perform future improvements such as making it easier to shrink the memory pool used by Kestrel.
+
+### IAsyncDisposable supported
+
+<xref:System.IAsyncDisposable> is now available for controllers, Razor Pages, and View Components. Asynchronous versions have been added to the relevant interfaces in factories and activators:
+
+* The new methods offer a default interface implementation that delegates to the synchronous version and calls <xref:System.IDisposable.Dispose%2A>.
+* The implementations override the default implementation and handle
+disposing `IAsyncDisposable` implementations.
+* The implementations favor `IAsyncDisposable` over `IDisposable` when
+both interfaces are implemented.
+* Extenders must override the new methods included to support
+`IAsyncDisposable` instances.
+
+<!-- TODO add ask @sharpcms to suggest When should I use IAsyncDisposable? and down  or copy contents from https://github.com/dotnet/AspNetCore.Docs/pull/23718 -->
```

Comment:
@JuergenGutsch can you copy/paste your [excellent IAsyncDisposable in MVC blog post](https://asp.net-hacker.rocks/2021/03/29/aspnetcore6-03-iasyncdisposable.html) to https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/release-notes/static/IAsyncDisposable.txt (edit in browser) ?
That way you get a commit credit and I'll migrate it to here. Or if it's easier  just add it as a suggestion here (but when I S&M, you won't get the commit credit)

Reply:
@Rick-Anderson 
Done. I copied the contents to the IAsyncDisposable.txt. I guess you'll need to rewrite it a little bit.
Anyway glad you asked me for the contribution. Thanks :-)